docs/streams/developer-guide/app-reset-tool.html

```diff
@@ -0,0 +1,236 @@
+
+    <span class="c1">// Delete the application&#39;s local state.</span>
+    <span class="c1">// Note: In real application you&#39;d call `cleanUp()` only under</span>
+    <span class="c1">// certain conditions.  See tip on `cleanUp()` below.</span>
+    <span class="n">app</span><span class="o">.</span><span class="na">cleanUp</span><span class="o">();</span>
+
+    <span class="n">app</span><span class="o">.</span><span class="na">start</span><span class="o">();</span>
+
+    <span class="c1">// Note: In real applications you would register a shutdown hook</span>
+    <span class="c1">// that would trigger the call to `app.close()` rather than</span>
+    <span class="c1">// using the sleep-then-close example we show here.</span>
+    <span class="n">Thread</span><span class="o">.</span><span class="na">sleep</span><span class="o">(</span><span class="mi">30</span> <span class="o">*</span> <span class="mi">1000L</span><span class="o">);</span>
+    <span class="n">app</span><span class="o">.</span><span class="na">close</span><span class="o">();</span>
+  <span class="o">}</span>
+
+<span class="o">}</span>
+</pre></div>
+                </div>
+                <p><strong>Tip:</strong> To avoid the corresponding recovery overhead, you should not call <code class="docutils literal"><span class="pre">cleanUp()</span></code> unconditionally and every time an
+                    application instance is restarted or resumed. For exammple, in a production application you could use  command line
+                    arguments to enable or disable the <code class="docutils literal"><span class="pre">cleanUp()</span></code> call on an as-needed basis.</p>
+                <p>You can then perform run-reset-modify cycles as follows:</p>
+                <div class="highlight-bash"><div class="highlight"><pre><span></span><span class="c1"># Run your application</span>
+$ bin/kafka-run-class io.confluent.examples.streams.ResetDemo
+
+<span class="c1"># After stopping all application instances, reset the application</span>
+$ bin/kafka-streams-application-reset --application-id my-streams-app <span class="se">\</span>
+                                      --input-topics my-input-topic <span class="se">\</span>
+                                      --intermediate-topics rekeyed-topic
+
+<span class="c1"># Now you can modify/recompile as needed and then re-run the application again.</span>
+<span class="c1"># You can also experiment, for example, with different input data without</span>
+<span class="c1"># modifying the application.</span>
+
+</pre></div>
+</div>
+</div>
+</div>
+</div>
+
+
+               </div>
+              </div>
+  <div class="pagination">
+    <a href="/{{version}}/documentation/streams/developer-guide/running-app" class="pagination__btn pagination__btn__prev">Previous</a>
```

Comment:
Should we consider showing up the content of the `prev` / `next` as well? For exmaple:

```
Previous: Running Streams Applications

Next: Streams Security
```

Reply:
@guozhangwang the current framework does not support longer titles in the Prev/Next buttons. For example, the additional text would look like:
![image](https://user-images.githubusercontent.com/11722533/33910226-d695da62-df42-11e7-9100-863603616530.png)